stdlib/config/config.lua

```diff
@@ -0,0 +1,152 @@
+    -----------------------------------------------------------------------
+    local Config = {}
+
+    --- Get a stored config value.
+    -- @param path [required] a string representing the variable to retrieve
+    -- @param default (optional) value to be used if path is nil
+    -- @return value at path or nil if not found and no default given
+    function Config.get(path, default)
+        if type(path) ~= "string" or path:is_empty() then error("path is invalid") end
+
+        local config = config_table
+
+        local c = testReservedCharacters(path)
+        if c ~= nil then error("path '" .. path .. "' contains the reserved character '" .. c .. "'") end
+
+        local pathParts = path:split('.')
+        local part = config;
+        local value = nil;
+
+        for key = 1, #pathParts, 1 do
+            if (type(part) ~= "table") then
+                value = nil;
+                break;
+            end
+
+            value = part[pathParts[key]];
+            part = part[pathParts[key]];
+        end
+
+        if (type(value) == "table") then
+            --Force break references.
+            return table.deepcopy(value);
+        elseif (value ~= nil) then
+            return value;
+        else
+            return default;
+        end
+    end
+
+    --- Set a stored config value.
+    -- @param path [required] a string, config variable to set
+    -- @param data (optional) to set path to. If nil it behaves identical to Config.delete()
+    -- @return boolean true on success; false on failure.
+    function Config.set(path, data)
+        if type(path) ~= "string" or path:is_empty() then error("path is invalid") end
+
+        local config = config_table
+
+        local c = testReservedCharacters(path)
+        if c ~= nil then error("path contains the reserved character '" .. c .. "'") end
+
+        local pathParts = path:split('.')
+        local part = config;
+        local value = nil;
+
+        for key = 1, #pathParts - 1, 1 do
+            if (type(part[pathParts[key]]) ~= "table") then
+                part[pathParts[key]] = {};
+            end
+
+            value = part[pathParts[key]];
+            part = part[pathParts[key]];
+        end
+
+        part[pathParts[#pathParts]] = data;
+
+        return 1;
+    end
+
+    --- Delete a stored config value.
+    -- @param path a string, config variable to set
+    -- @return boolean, true on success, false on failure.
+    function Config.delete(path)
+        if type(path) ~= "string" or path:is_empty() then error("path is invalid") end
+
+        local config = config_table
+
+        local c = testReservedCharacters(path)
+        if c ~= nil then error("path contains the reserved character '" .. c .. "'") end
+
+        local pathParts = path:split('.')
+        local part = config
+        local value = nil
+
+        for key = 1, #pathParts - 1, 1 do
+            if (type(part[pathParts[key]]) ~= "table") then
+                return 0
```

Comment:
Aw crap. The return values are the ones that @dkaisers wanted. So that later when Config is extended and can return multiple results existing code shouldn't need updated. I forgot to update the docs to reflect this change.